### ISSUE TITLE
Set plan name when creating plan

### ISSUE THREAD

**Issue by prenagha** [pgMustard/pgMustard#68]
It would be useful to set the plan "name" on the same first screen where you paste the plan contents.

**Comment by prenagha** [comment]
yes I know name and likely share level from the start. and now that I know the share level can be changed... would be good to do that on the first screen.

**Comment by prenagha** [comment]
also its nice to show the share level at the start since it is less worrisome when the share level is explicitly shown

**Comment by mchristofides** [comment]
Wonderful, that makes sense. 

Interesting point on the worry front — do you mean for first time users, or for yourself?

**Comment by prenagha** [comment]
worry is both. i wouldn't want to expose our queries ever, so its always nice to see it explicitly set at the point of first saving.

**Comment by mchristofides** [comment]
@prenagha thanks again for this feedback. As of yesterday, we now show "Untitled plan" along with a button to add a name to all plans that are not yet named. The idea is to nudge folks who want to add a name early on, without adding more things to think about on that first page. I realise it isn't exactly what you want, but I hope it's better for you than what we had before.

![untitled-plan](https://github.com/pgMustard/pgMustard/assets/10532911/152bacfd-7592-48d0-aa77-775400c62cea)
